tests/compile-fail/copies.rs

```diff
@@ -249,6 +249,21 @@ fn if_same_then_else() -> Result<&'static str, ()> {
         -0.0
     };
 
+    // Different NaNs
+    let _ = if true {
+        1.0 / 0.0
+    } else {
+        (-5f32).sqrt()
```

Comment:
`(-5f32).sqrt()` is not a constant expression and `1.0/0.0` is infinity, not NaN. So it's not affected by that bug anyway.
`0.0 / 0.0` vs. `std::f32::NAN` would have been more relevant.